File: primeNgTest/src/app/viewSettings.ts
import { SortMeta, FilterMetadata } from "primeng/api";

export interface ViewSettings {
    /** server-issued id of this view */
    id: number;
    listId: number;

    name: string;
    isPublic: boolean;
    isTemporary: boolean;
    isDefault: boolean;
    
    /** Visible columns and their order */
    columns: string[];

    /** Relative (to the whole table) widths of individual columns, in their order */
    columnRelativeWidths?: number[];

    sort?: SortMeta[];

    filters: { [s: string]: FilterMetadata; }

}

File: McvAngularTest2/Models/ViewSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace McvAngularTest2.Models
{
    public class ViewSettings : StoreableViewSettings
    {
        public int id { get; set; }
        public int listId { get; set; }

        public string name { get; set; }
        public bool isPublic { get; set; }
        public bool isTemporary { get; set; }
        public bool isDefault { get; set; }


    }
}